### ISSUE TITLE
Using signs outside ASCII in response body

### ISSUE THREAD

**Issue by dominikjeske** [pact-foundation/pact-net#468]
I'm testing pact-net and I spotted weird behavior when using signs outside ASCII table. I was using my native letters from my country Poland (like ł,ó,ń) - they are UTF-8 compatible but when I'm using them in WithJsonBody mock that is used under the cover is returning empty body. I cannot find in documentation is this by design or is it bug. I suspect some but when interop with native library is used.

**Comment by adamrodger** [comment]
Yeah that sounds like interop problems would be the first thing to check.
We should definitely have some tests using non-ASCII characters and we
don't, so that's the first thing to add.

**Comment by adamrodger** [comment]
I've tested this out locally and it breaks pretty badly with non-ASCII characters in things like consumer/provider names, interaction names, provider states, etc. It generally seems to fail everywhere because the FFI reports invalid UTF-8.

That's a bit of a problem given C# strings are UTF-16 and the P/Invoke options are just "Unicode". We may have to marshall strings manually or something.

You also get completely garbled responses back when you try to use the Unicode option to retrieve things like server logs:

![image](https://github.com/pact-foundation/pact-net/assets/508586/8d827301-82eb-4742-b476-6526a4942127)

The difference is that I get errors whereas you report that the test passes but then silently contains no data. I've not managed to reproduce that, although I could definitely see that being a possibility.

I can only apologise for that, and recommend that in the meantime you only use ASCII characters until the issue can be resolved.

**Comment by adamrodger** [comment]
The linked PR fixes the issues on Windows, but is writing unreadable chars on Linux, so not sure what's happening there.
